sycl/source/detail/event_impl.cpp

```diff
@@ -95,6 +103,33 @@ event_impl::event_impl(QueueImplPtr Queue) : MQueue(Queue) {
 
 void event_impl::wait(
     std::shared_ptr<cl::sycl::detail::event_impl> Self) const {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  xpti::trace_event_data_t *TraceEvent = nullptr;
+  int32_t MStreamID = xptiRegisterStream(SYCL_STREAM_NAME);
+  std::string Name;
+  if(xptiTraceEnabled()) {
+    // Create a string with the event address so it
+    // can be associated with other debug data
+    xpti::utils::StringHelper SH;
+    Name = SH.nameWithAddress("event.wait", (void *)MEvent);
+
+    // We can emit the wait associated with the graph if the 
+    // event does not have a command object or associated with
+    // the command object, if it exists
+    if(MCommand) {
+      Command *Cmd = (Command *)MCommand;
+      TraceEvent = ( Cmd->MTraceEvent ) ? (xpti_td *)Cmd->MTraceEvent : g_sycl_graph_event;
```

Comment:
@tovinkere I think @keryell means something like this:

```suggestion
      TraceEvent = Cmd->MTraceEvent ? (xpti_td *)Cmd->MTraceEvent : g_sycl_graph_event;
```

Reply:
Done!